File: backend/src/main/java/es/udc/paproject/backend/model/services/TeamService.java
package es.udc.paproject.backend.model.services;

import java.util.List;

import es.udc.paproject.backend.model.entities.Season;
import es.udc.paproject.backend.model.entities.Team;
import es.udc.paproject.backend.model.exceptions.InstanceNotFoundException;

public interface TeamService {
    //done
    Team addTeam(Long userId, Team team) throws InstanceNotFoundException;
    //done
    void addTeamToSeason(Long seasonId, Team team, Long userId) throws InstanceNotFoundException;
    //done
    Team findTeamById(Long userId,Long teamId) throws InstanceNotFoundException;
    //done
    Team findTeamByName(Long userId, String teamName) throws InstanceNotFoundException;
    //done
    List<Team> findAllTeams(Long userId) throws InstanceNotFoundException;
    
    List<Season> findSeasonsToTeam(Long userId, Long teamId) throws InstanceNotFoundException;
    //done
    void removeTeam(Long userId, Long teamId) throws InstanceNotFoundException;
    //done
    Team updateTeam(Long userId, Team team) throws InstanceNotFoundException;
}

File: frontend/src/backend/teamService.js
import {config, appFetch} from './appFetch';

export const findAllTeams = (onSuccess) =>
    appFetch('/teams', config('GET'), onSuccess);

export const findTeamById = (id, onSuccess) =>
    appFetch(`/teams/team/${id}`, config('GET'), onSuccess);

export const findTeamByName = (name, onSuccess) =>{
    appFetch(`/teams/name/${name}`, config('GET'), onSuccess);
}

export const addTeam = (name, onSuccess, onErrors) =>
    appFetch(`/teams/addTeam/?name=${name}`, config('POST'), onSuccess, onErrors);

export const updateTeam = (id, name, onSuccess, onErrors) =>
    appFetch(`/teams/update/${id}?name=${name}`, config('PUT'), onSuccess, onErrors);

export const removeTeam = (id, onSuccess, onErrors) =>{
    appFetch(`/teams/remove/${id}`, config('DELETE'), onSuccess, onErrors);
}

export const addTeamToSeason = (seasonId, teamId, onSuccess, onErrors) =>
    appFetch(`/teams/addTeamtoSeason/${seasonId}?teamId=${teamId}`, config('POST'), onSuccess, onErrors);

export const findTeamsToSeason = (seasonId, onSuccess, onErrors) => 
    appFetch(`/teams/toSeason/${seasonId}`, config('GET'), onSuccess, onErrors);

